### ISSUE TITLE
Provide model definition when using ref override

### ISSUE THREAD

**Issue by rodik** [innoverio/vscode-dbt-power-user#92]
Hi,

This is a suggestion for the model definition provider.

Sometimes an override is used for `ref` in dbt projects and it would be really nice to have the model definition shortcut working in such cases. 

Example override:
```
{% macro ref_no_db(model_name) %}
  {% do return(builtins.ref(model_name).include(database=false)) %}
{% endmacro %}
```

Suggested solution: override macro must start with "ref_" to simplify parsing.

**Comment by mdesmet** [comment]
I also use this ref override in some of my projects, from what i remember it just removes the database when rendering the SQL. I would think the model definition shortcut shouldn't depend on it.

Could you explain the issue you're experiencing?

**Comment by rodik** [comment]
Hi and sorry for not being clear enough with original comment. 

When the reference function is not named `ref` the "Go to definition" feature doesn't work on referenced models.

Example: 
Our override is called `ref_no_db` and model definition for *my_model* isn't available because of this when referenced:

`select * from {{ ref_no_db('my_model') }}`

Can't use F12 or Ctrl+Click to reach *my_model* file from above code chunk. Works fine when I switch back to `ref`. 
Would be great if functionality could also work with any custom `ref_*` function.

**Comment by mdesmet** [comment]
You should read following doc: https://docs.getdbt.com/reference/dbt-jinja-functions/builtins#usage

Just rename your override ref_no_db to ref, and the extension should work, that's how I do it in other projects. :)

**Comment by rodik** [comment]
Yes, I tried that but it caused issues with other stuff in our project so we had to keep both original ref and ref_no_db. 
If I remember correctly, the trouble with calling the override "ref" caused some internal (metadata) dbt queries to fail on Snowflake as a 3-part name was expected in such cases.

**Comment by mdesmet** [comment]
Also using snowflake in some projects, no issues with the override.

```
{% macro ref(modelname) %}{% do return(builtins.ref(modelname).include(database=False)) %}{% endmacro %}
```

**Comment by rodik** [comment]
Try using stuff like 
`{{ dbt_utils.star( ref('some_model') ) }}`
with the ref override named `ref`

This will fire a `describe` statement on Snowflake without a 3-part name and throw an error. I know this exact error can be solved in other ways (like using default_namespace param for users) but most just create new issues.

To generalize - if a package functionality depends on 3-part names you brake those by naming the override `ref`. Currently we would write:

```
select  
    {{ dbt_utils.star( ref('some_model') ) }}
from {{ ref_no_db('some_model') }}
```

**Comment by mdesmet** [comment]
If your overriden ref is not smart enough, then make it smarter: You can get the current scheme and current database and compare it towards the input, if it's not the same you keep the database.

**Comment by rodik** [comment]
That is a good suggestion. Will try.

**Comment by rodik** [comment]
This is the override we ended up with: 
``` jinja
{% macro ref(model_name) %}

  {% if config.get('materialized') == "view" %}
    {% do return(builtins.ref(model_name).include(database=false)) %}
  {% else %}
    {% do return(builtins.ref(model_name).include(database=true)) %}
  {% endif %}

{% endmacro %}
```
Views are really the only type of model where 2-part names are needed.  Everything else defaults to original ref and 3-part names.
